### PR TITLE
Update katas to use local variable type inference.

### PR DESCRIPTION
**Pull request by donraab**
Signed-off-by: Donald Raab <Donald.Raab@bnymellon.com>

### CODE REVIEW

File: company-kata/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java [hunk 1]
```diff
@@ -95,7 +97,9 @@ public void filterOrderValues()
         List<Order> orders = this.company.getMostRecentCustomer().getOrders();
```
Comment (prathasirisha) [inline company-kata/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java:97]: Was this deliberately not updated to "var" 
var orders = this.company.getMostRecentCustomer().getOrders()

Comment (donraab) [reply company-kata/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java:97]: Yes, I left this call typed so the user has the specific type information available so they known which APIs they have to use in their code.

File: company-kata/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java [hunk 2]
```diff
@@ -107,7 +111,9 @@ public void filterOrderValuesUsingPrimitives()
...
```
[the remark this answers is not in the document]
Comment (donraab) [reply company-kata/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java:111]: I tried to limit usage of var to expected values that the developer will not need to call APIs on the type.

File: company-kata/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java [hunk 3]
```diff
@@ -118,6 +124,8 @@ public void filterOrders()
     {
         List<Order> orders = this.company.getMostRecentCustomer().getOrders();
```
[the remark this answers is not in the document]
Comment (donraab) [reply company-kata/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java:125]: Same comment as above... I tried to limit usage of var to expected values that the developer will not need to call APIs on the type.

**Comment by donraab** [comment]
Thanks for the code review @prathasirisha!  I will merge now.

**Pull request merged**